net/http/transport.go

```diff
@@ -20,28 +31,34 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	var (
 		err      error
 		body     []byte
-		backend  *backend
+		backend  *regbalancer.Backend
 		response *http.Response
 		service  = req.URL.Host
 		maxRetry = t.MaxRetry
+		balancer = t.Balancer
 	)
 	if maxRetry == 0 {
 		maxRetry = DefaultMaxRetry
 	}
+	if balancer == nil {
+		balancer = regbalancer.Default()
+	}
 	if req.Body != nil {
 		body, _ = ioutil.ReadAll(req.Body)
 	}
 	for i := 0; i <= maxRetry; i++ {
-		if backend, err = _balancer.next(service, t.MaxRequestsByBackend); err == nil {
-			backend.incConcurrentRequest(1)
-			defer backend.incConcurrentRequest(-1)
+		if backend, err = balancer.Next(service, t.MaxRequestsByBackend); err == nil {
+			backend.IncConcurrentRequest(1)
```

Comment:
IMHO, this methods shouldn't be public. If you need to use is in different packages, then you can replace:
```
backend.IncConcurrentRequest(1)
defer backend.IncConcurrentRequest(-1)
[...code...]
```
To something like
```
backend.ProcessRequest(req)
```

Reply:
I think that it's not possible to make effectively for HTTP and GRPC this way